service/controller/v13/cluster_resource_set.go

```diff
@@ -110,15 +114,26 @@ func NewClusterResourceSet(config ClusterResourceSetConfig) (*controller.Resourc
 		return nil, microerror.Maskf(invalidConfigError, "%T.APIWhitelist.SubnetList must not be empty when %T.APIWhitelist is enabled", config)
 	}
 
-	var encrypter *kms.Encrypter
-	{
+	var encrypterBackend encrypter.Interface
+	if config.Encrypter == "vault" {
```

Comment:
When we have multiple choices we should explicitly handle all and have a default to fail in case wrong settings are given. Assuming arbitrary values imply a specific configuration can easily lead to problems down the road.

Reply:
Sure, done. PTAL